pkg/plugin/processor/builtin/unwrap/unwrap_opencdc.go

```diff
@@ -0,0 +1,293 @@
+	}
+
+	opencdcRec, err := u.unmarshalRecord(data)
+	if err != nil {
+		return sdk.ErrorRecord{Error: cerrors.Errorf("failed unmarshalling record: %w", err)}
+	}
+	opencdcRec.Position = rec.Position
+
+	// Position is the only key we preserve from the original record to maintain the reference respect other messages
+	// that will be coming from in the event of chaining pipelines (e.g.: source -> kafka, kafka -> destination)
+	return sdk.SingleRecord(opencdcRec)
+}
+
+func (u *openCDCProcessor) Teardown(context.Context) error {
+	return nil
+}
+
+func (u *openCDCProcessor) unmarshalRecord(structData opencdc.StructuredData) (opencdc.Record, error) {
+	operation, err := u.unmarshalOperation(structData)
+	if err != nil {
+		return opencdc.Record{}, cerrors.Errorf("failed unmarshalling operation: %w", err)
+	}
+
+	metadata, err := u.unmarshalMetadata(structData)
+	if err != nil {
+		return opencdc.Record{}, cerrors.Errorf("failed unmarshalling metadata: %w", err)
+	}
+
+	key, err := u.unmarshalKey(structData)
+	if err != nil {
+		return opencdc.Record{}, cerrors.Errorf("failed unmarshalling key: %w", err)
+	}
+
+	payload, err := u.unmarshalPayload(structData)
+	if err != nil {
+		return opencdc.Record{}, cerrors.Errorf("failed unmarshalling payload: %w", err)
+	}
+
+	return opencdc.Record{
+		Key:       key,
+		Metadata:  metadata,
+		Payload:   payload,
+		Operation: operation,
+	}, nil
+}
+
+// unmarshalOperation extracts operation from a structuredData record.
+func (u *openCDCProcessor) unmarshalOperation(structData opencdc.StructuredData) (opencdc.Operation, error) {
+	var operation opencdc.Operation
+	op, ok := structData["operation"]
+	if !ok {
+		return operation, cerrors.New("no operation")
+	}
+
+	switch opType := op.(type) {
+	case opencdc.Operation:
+		operation = opType
+	case string:
+		if err := operation.UnmarshalText([]byte(opType)); err != nil {
+			return operation, cerrors.Errorf("invalid operation %q", opType)
+		}
+	default:
+		return operation, cerrors.Errorf("expected a opencdc.Operation or a string, got %T", opType)
+	}
+	return operation, nil
+}
+
+// unmarshalMetadata extracts metadata from a structuredData record.
+func (u *openCDCProcessor) unmarshalMetadata(structData opencdc.StructuredData) (opencdc.Metadata, error) {
+	var metadata opencdc.Metadata
+	meta, ok := structData["metadata"]
+	if !ok {
+		return metadata, cerrors.New("no metadata")
+	}
+
+	switch m := meta.(type) {
+	case opencdc.Metadata:
+		metadata = m
+	case map[string]interface{}:
+		metadata = make(opencdc.Metadata, len(m))
+		for k, v := range m {
+			metadata[k] = fmt.Sprint(v)
+		}
+	default:
+		return nil, cerrors.Errorf("expected a opencdc.Metadata or a map[string]interface{}, got %T", m)
+	}
+
+	return metadata, nil
+}
+
+// unmarshalKey extracts key from a structuredData record.
+func (u *openCDCProcessor) unmarshalKey(structData opencdc.StructuredData) (opencdc.Data, error) {
```

Comment:
Actually, why don't we just reuse `convertPayloadData`? I see the only difference is that the key is apparently required. Why though? It's not really a required field.